src/todt.c

```diff
@@ -62,15 +60,15 @@ Symbol *toSymbolCppTypeInfo(ClassDeclaration *cd);
 
 /* ================================================================ */
 
-dt_t **Initializer_toDt(Initializer *init, dt_t **pdt)
+void Initializer_toDt(Initializer *init, DtBuilder& dtb)
 {
     class InitToDt : public Visitor
     {
     public:
-        dt_t **pdt;
+        DtBuilder dtb;
 
-        InitToDt(dt_t **pdt)
-            : pdt(pdt)
+        InitToDt(DtBuilder& dtb)
+            : dtb(dtb)
```

Comment:
The interior pointer + the default constructor have created another problem for conversion.

The usual way to translate a default ctor into D is to use static opCall:
`auto dtb = DtBuilder();`

But this will break the interior pointer.  Any suggestions?

@WalterBright Why is it so hard to stick to the common subset and make life easier for the person converting the code?  I don't understand why you repeatedly insist on deviating from the existing feature use of a compiler you wrote, and the common subset shared with a language you designed.

Conversion to D is straightforward once the non-common-subset issues are resolved.  https://github.com/yebblies/dmd/commit/3aff82e0c63ec71d2670538afc5e40bbc1719b13